File: app/src/main/java/net/itsjustsomedude/tokens/db/CoopDao.java
package net.itsjustsomedude.tokens.db;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;
import androidx.room.Upsert;

import java.util.List;

@Dao
public interface CoopDao {
	@Insert
	void insert(Coop coop);

	@Update
	void update(Coop coop);

	@Upsert
	void upsert(Coop coop);

	@Delete
	void delete(Coop coop);

	@Query("DELETE FROM Coop")
	void deleteAllCoops();

	@Query("SELECT * FROM Coop ORDER BY id ASC")
	List<Coop> getAllCoops();
}

File: app/src/main/java/net/itsjustsomedude/tokens/db/CoopDao.kt
package net.itsjustsomedude.tokens.db

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update

@Dao
interface CoopDao {
    @Insert
    suspend fun insert(coop: Coop): Long

    @Update
    suspend fun update(coop: Coop)

    @Delete
    suspend fun delete(coop: Coop)

    @Query("DELETE FROM Coop WHERE id = :id")
    suspend fun deleteById(id: Long)

    @Query("SELECT * FROM Coop ORDER BY id DESC")
    fun listCoopsLiveData(): LiveData<List<Coop>>

    @Query("SELECT * FROM Coop WHERE id = :id")
    fun getCoopLiveData(id: Long): LiveData<Coop?>

    @Query("SELECT * FROM Coop WHERE name = :name AND contract = :kevId")
    fun getCoopLiveDataByName(name: String, kevId: String): LiveData<Coop?>

    @Query("SELECT * FROM Coop WHERE name = :name AND contract = :kevId")
    suspend fun getCoopByName(name: String, kevId: String): Coop?

    @Query("SELECT * FROM Coop WHERE id = :id")
    suspend fun getCoop(id: Long): Coop?
}